src/master/allocator/mesos/slavesorter/cpu_first/slavesorter.cpp

```diff
@@ -162,7 +162,28 @@ bool ResourceSlaveSorterCPUFirst::isOfferable(
   Resources minOfferableResources = minOfferable.at(role);
   double minCpusOfferable = minOfferableResources.cpus().getOrElse(0);
   double offerCpus = resources.cpus().getOrElse(0);
```

Comment:
FAR/Better: not sure if achievable easily, but we should iterate over each "key" resource to be more generic maybe

Reply:
it's not easy because the type of each is not always the same (I've just discovered this)